ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -189,52 +187,59 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
-                    take(1)
-                ).subscribe(_ => {
-                    this.isLoaded$.next(true);
-                    this.currentSectionIndex = nextIndex;
-                    this.visitedSectionIndexes[nextIndex] = true;
-                    this.saveLastVisitedSectionIndex(nextIndex);
-                });
-            }
+            this.submitAnnouncementService.announceSubmit();
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(() => {
+                            // run validations and compute flags for blocks to enable scrolling to errors
+                            this.currentSection.validate();
+                            this.model && this.model.shouldScrollToFirstInvalidQuestion();
+                            // triggers scrolling
+                            this.validationRequested = true;
+                        }),
+                        // delay needed for validationRequested to be processed
+                        delay(0),
+                        tap(() => {
+                            this.sendSectionAnalytics();
+                            // reset scrolling signal
+                            this.validationRequested = false;
+                            if (this.currentSection.valid) {
+                                this.visitedSectionIndexes[nextIndex] = true;
+                                this.saveLastVisitedSectionIndex(nextIndex);
+                                this.currentSectionIndex = nextIndex;
+                                if (scroll) {
+                                    this.scrollToTop();
+                                }
+                            }
+                        })
+                    )
+                ),
+                take(1)
+            ).subscribe();
         }
     }
 
     public decrementStep(scroll: boolean = true): void {
         const previousIndex = this.previousAvailableSectionIndex();
         if (previousIndex !== -1) {
-            // if we move forwards or backwards, let's reset our validation display
-            this.resetValidationState();
-
             if (scroll) {
                 this.scrollToTop();
             }
-
-            this.isPageBusy.pipe(
-                tap(_ => this.isLoaded$.next(false)),
+            this.isPageBusy.pipe(startWith(true)).pipe(
+                filter(pageIsBusy => !pageIsBusy),
                 debounceTime(this.timeToDebounce),
-                filter(busy => !busy),
+                tap(() => {
+                    // if we move forwards or backwards, let's reset our validation display
+                    this.resetValidationState();
+                    this.isLoaded$.next(true);
```

Comment:
Seems like it should be eliminated as well ( as you removed all `isLoaded$` in `incrementStep`/`decrementStep`)

Reply:
Yup. Looks like it is not needed. Removed: e96b37efb